### ISSUE TITLE
Sentry.AspNetCore.Sentry middleware uses types that aren't present in .NET Core 3

### ISSUE THREAD

**Issue by alexeyzimarev** [getsentry/sentry-dotnet#311]
Please mark the type framework used:

- [x ] ASP.NET Core

Please mark the type of the runtime used:

- [x ] .NET Core

* Version: 3.0

Please mark the NuGet packages used:

- [ x] Sentry.AspNetCore

Sentry AspNetCore middleware crashes for .NET Core Web SDK application unconditionally:

```
System.TypeLoadException: Could not load type 'Microsoft.AspNetCore.Http.Internal.BufferingHelper' from assembly 'Microsoft.AspNetCore.Http, Version=3.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'.
   at async Task Sentry.AspNetCore.SentryMiddleware.InvokeAsync(HttpContext context)
   at void System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start<TStateMachine>(ref TStateMachine stateMachine)
   at void System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start<TStateMachine>(ref TStateMachine stateMachine)
   at Task Sentry.AspNetCore.SentryMiddleware.InvokeAsync(HttpContext context)
   at Task Microsoft.AspNetCore.Hosting.HostingApplication.ProcessRequestAsync(Context context)
   at async Task Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests<TContext>(IHttpApplication<TContext> application)
```

**Comment by bruno-garcia** [comment]
Thanks for raising this @alexeyzimarev 
Is this happening with the latest beta? `2.0.0-beta7`.
If so, could you please provide more details to help us reproduce the issue?

**Comment by alexeyzimarev** [comment]
No, it’s 1.2, the latest GA. We use Sentry in production and we have a policy to only use GA in production. The .NET Core 3 is GA for a couple of months so we expected Sentry libs to support it.

In addition, I found out that even using Sentry.Serilog somehow adds Sentry.AspNetCore references, which is quite unexpected.

**Comment by bruno-garcia** [comment]
> No, it’s 1.2, the latest GA. We use Sentry in production and we have a policy to only use GA in production. The .NET Core 3 is GA for a couple of months so we expected Sentry libs to support it.

They've introduced a breaking change and fixed it in 2.0.0-beta. It's true we're taking a while to ship 2.0.0 but we've just been busy.

Please feel free to use 2.0.0-beta, the reason beta wasn't dropped yet is that we want to fit more breaking changes to it.

> In addition, I found out that even using Sentry.Serilog somehow adds Sentry.AspNetCore references, which is quite unexpected.

Sorry but can you elaborate how this happens? `Sentry.Serilog` is completely decoupled to `Sentry.AspNetCore`. If the app has both, the do cooperate.

**Comment by R10-Rasmus** [comment]
This is still an issue:
I use:

[x ] ASP.NET Core
Please mark the type of the runtime used:

[x ] .NET Core
Version: 3.1
Please mark the NuGet packages used:

[ x] Sentry 2.0.0-beta8 

My code is:
        public static IHostBuilder CreateHostBuilder(string[] args) =>
            Host.CreateDefaultBuilder(args)
                .ConfigureWebHostDefaults(webBuilder =>
                {
                     webBuilder.UseSentry();
                     webBuilder.UseStartup<Startup>();
                });

When I try to call a controller method I get:
Microsoft.AspNetCore.Server.IIS.Core.IISHttpServer: Error: Connection ID "18374686502757072928", Request ID "80000021-0005-ff00-b63f-84710c7967bb": An unhandled exception was thrown by the application.

System.TypeLoadException: Could not load type 'Microsoft.AspNetCore.Http.Internal.BufferingHelper' from assembly 'Microsoft.AspNetCore.Http, Version=3.1.1.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'.
   at Sentry.AspNetCore.SentryMiddleware.InvokeAsync(HttpContext context)
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[TStateMachine](TStateMachine& stateMachine)
   at Sentry.AspNetCore.SentryMiddleware.InvokeAsync(HttpContext context)
   at Microsoft.AspNetCore.HostFiltering.HostFilteringMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Hosting.HostingApplication.ProcessRequestAsync(Context context)
   at Microsoft.AspNetCore.Server.IIS.Core.IISHttpContextOfT`1.ProcessRequestAsync()
Microsoft.AspNetCore.Hosting.Diagnostics: Information: Request finished in 10.2048ms 500

**Comment by bruno-garcia** [comment]
Please use the latest beta package. It's fixed there.

**Comment by rfink** [comment]
Getting this issue in .Net core 3.1, with version 2.1.2-beta

**Comment by bruno-garcia** [comment]
@rfink The original issue was using Sentry 1.2.0 which is NOT compatible with .NET Core 3.1.

[This comment](https://github.com/getsentry/sentry-dotnet/issues/311#issuecomment-577040682) is a different issue (uses Sentry 2.x and has different stack trace) which was raised on a separate issue #345 that was resolves.

I'm going to lock this issue since it's a mess and the issue was resolved (Sentry 1.2 isn't compatible with .NET Core 3).

Please raise a new issue with details of what's going on and ideally share a link to a github repository which I can run and reproduces this issue.
